### ISSUE TITLE
Thermal Zone tab crash when editing zone equipment

### ISSUE THREAD

**Issue by MatthewSteen** [NREL/OpenStudio#1608]
To reproduce in 1.7.1:

1. create several thermal zones
2. add zone equipment to each (tested on FCUs and WSHPs) 
3. move through the zone equipment like you're trying to edit each one
4. if you have one selected, select another, and then click the IG scroll bar before the app has had time to switch to the next one.

At this point the app will close or crash (closed for me on a new model, crashed on a existing model).

**Comment by eringold** [comment]
@kbenne I also get this error (with OSApp v1.7.0, using FCU objects), but with the following additions to @MatthewSteen's list:

1. crash typically occurs after 6-12 cycles between selecting zone equipment and clicking the scroll bar before the object edit information loads (however, I have had it happen after only one or two cycles). 
2. crash can occur if the selection cycle is between just one equipment instance and the scroll bar, not only when selecting another equipment object. 
3. crash doesn't occur at the moment the scroll bar is clicked, but happens after a delay that seems to correspond with the object information load time.

**Comment by kbenne** [comment]
OK.  I'll get to the bottom of this.  Probably related to the recent grid view feature.

**Comment by MatthewSteen** [comment]
Also occurred when deleting PTACs in 1.7.5.

**Comment by kbenne** [comment]
@MatthewSteen and @eringold I tested this on my Mac with the official 1.7.0 package and I was able to reproduce the issue.

I built a debug build of current develop and was not able to reproduce.  That is not absolute proof that the bug is gone.  My ability to reproduce amounted to rapid fire clicking through various zone equipment AND jumping over to scroll quickly before the inspector is loaded.  But it is one of those things (at least from my perception) that is sort of imprecise to reproduce.  I will not guarantee that even develop is crash proof with respect to this issue.  One thing is that these kinds of issues (basically involving the application's event loop) can be platform specific, and maybe even debug / release build specific.  What I'm hoping to do is generate up an installer package of current develop build on Windows and ask you to install it and verify that the issue is gone.  Does that sound reasonable?

**Comment by kbenne** [comment]
I know @evanweaver has done some work in this area.  @evanweaver if you can point to a specific commit that you think addresses this issue, that would be nice for record keeping.

**Comment by kbenne** [comment]
Note I just saw @MatthewSteen's comment about 1.7.5.  That doesn't look promising.  There are some commits between then and now but not many that appear related.

**Comment by evanweaver** [comment]
Sorry @kbenne, I've been in gridview feature branches.  I haven't seen the issue, but I'm not aware that I may have inadvertently fixed it.

**Comment by MatthewSteen** [comment]
FYI @kbenne this occurred in a large model after applying the 90.1 System 1 measure. I was in the process of deleting PTACs from the secondary zones.

**Comment by kbenne** [comment]
Yeah its defiantly still a thing.  I think @eringold's comments are key.  It seems that if you want to recreate this issue it helps to rapid fire select several times (6-12 cycles seems about right) between zone equipment then quickly jump to IG. We'll get this tracked down. Two or three cycles doesn't seem to do it for me.

**Comment by eringold** [comment]
@kbenne I'm happy to test a develop build if you put together an installer.

**Comment by kbenne** [comment]
OK @eringold @MatthewSteen,  @evanweaver and I have spent a lot of time on this issue now.  Like I said I can confirm that it is definitely an issue.  I have now reproduced it on Mac and Windows.  The problem is that it has taken us a very long time clicking around in the manner described to reproduce.  I cannot identify an exact sequence that does it.  I know you gotta be fast.  (or at least in my experience) The problem for us is that even if we do have a potential fix, we aren't sure that we will know it is a fix because the issue is so highly intermittent.  Here this morning we worked through the debugger at least an hour doing extensive monkey clicking and didn't crash at all.  (Yesterday I did get a crash) Unless you are seeing much higher failure rates we are thinking we are going to close or at least move this way down the severity level.  

One of the reasons that we are inclined to this decision is that @evanweaver is working on a branch with extensive changes (simplifications) to the grid architecture and those changes will certainly touch the area of this issue.  So if we do take no action now, we can revisit once Evan merges.

I do have a (sort of wild) guess for a fix in mind which I ran by Evan.  The problem is like I said, I'm not sure we will know if it actually fixed anything because our failure rate has been so low.  And the fix itself has reasonable likelihood of unintended consequences.  If your failure rates are higher I am willing to try the idea I had in mind and send you a build to test out, but like I said unless this is a much more prevalent issue for you than I am seeing, I'd rather take no action at this time.

Bottom line, I think I'm asking you for input on how easy it is for you to make this happen. and if you consider it a major issue.  Unless you speak up strongly I'm going to adjust severity and put this on hold for Evan's work.

**Comment by MatthewSteen** [comment]
@kbenne sounds reasonable to me. I haven't been able to reproduce it in the past day deleting PTACs, even with some :monkey_face: clicking.

**Comment by kbenne** [comment]
For record here is my best guess at a fix.

https://github.com/NREL/OpenStudio/commit/b7a931dce05488bf7c31988f1a8c565420dbc883

Now that I talked everyone off this issue, I actually couldn't help myself from rolling a package.  I'll send out a download link for testing if anyone is so inclined.  The route I actually went with in that commit is pretty low risk so if you think it clears up the issue it might be worth considering for 1.8.

**Comment by kbenne** [comment]
@MatthewSteen I sent you a link to a package for testing.

@eringold I don't think I have your email, but if you send me a note I'll pass the link along to you too.

**Comment by MatthewSteen** [comment]
@kbenne received, I'll pass it along to @eringold.

**Comment by eringold** [comment]
@kbenne unfortunately I still get a crash with your build, but it's not the same 'hard' crash I was experiencing. I'm sending you an email with more info and a link to files.

**Comment by kbenne** [comment]
Anyone have a model that still does this?

**Comment by MatthewSteen** [comment]
I was not able to recreate in a 1.9.0 model by trying the following:

* cycling through radiant systems and deleting them
* cycling through and scrolling down in the IG
